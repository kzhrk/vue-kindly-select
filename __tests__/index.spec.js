import KindlySelect from '../src';
import Vue from 'vue/dist/vue';

describe(`vue-kindly-select`, () => {
  // set components
  Vue.use(KindlySelect);

  async function createVm() {
    const vm = new Vue({
      el: '#app'
    });

    await Vue.nextTick();

    return vm.$children[0];
  }

  beforeEach(() => {
    document.body.innerHTML = `
      <div id="app">
        <kindly-select name="select1">
          <kindly-option text="UK" value="1">
            <i>ðŸ‡¬ðŸ‡§</i>United Kingdom
          </kindly-option>
          <kindly-option text="USA" value="2">
            <i>ðŸ‡ºðŸ‡¸</i>United States of America
          </kindly-option>
          <kindly-option text="PRC" value="3">
            <i>ðŸ‡¨ðŸ‡³</i>People's Republic of China
          </kindly-option>
        </kindly-select>
      </div>
    `;
  });

  it('can mount', async () => {
    await createVm();

    expect(document.body.innerHTML).toMatchSnapshot();
  });

  it('select first option', async () => {
    const select = await createVm();

    [...select.$children][0].handleClick();

    expect(select.activedescendant).toBe([...select.$children][0].computedId);
    expect(select.buttonText).toBe([...select.$children][0].text);
    expect(select.selected).toBe([...select.$children][0].value);
  });

  it('close ul', async () => {
    const select = await createVm();
    const button = document.querySelector('button');

    button.click();
    expect(select.hidden).toBe('false');

    [...select.$children][0].handleClick();
    expect(select.hidden).toBe('true');

    button.click();
    expect(select.hidden).toBe('false');

    document.body.click();
    expect(select.hidden).toBe('true');
  });
});
