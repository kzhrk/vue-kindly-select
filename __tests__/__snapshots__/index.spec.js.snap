// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue-kindly-select can mount 1`] = `
"
      <div id=\\"app\\"><div class=\\"kindly-select\\"><button type=\\"button\\" role=\\"button\\" aria-haspopup=\\"listbox\\" class=\\"kindly-select-button\\"></button> <select name=\\"select1\\" aria-hidden=\\"true\\" aria-haspopup=\\"listbox\\" class=\\"kindly-select-el\\"><option value=\\"1\\">UK</option><option value=\\"2\\">USA</option><option value=\\"3\\">PRC</option></select> <ul aria-hidden=\\"true\\" role=\\"listbox\\" tabindex=\\"-1\\" class=\\"kindly-select-list\\"><li id=\\"select1--1\\" role=\\"option\\" class=\\"kindly-option\\"><i>ğŸ‡¬ğŸ‡§</i>United Kingdom
          </li> <li id=\\"select1--2\\" role=\\"option\\" class=\\"kindly-option\\"><i>ğŸ‡ºğŸ‡¸</i>United States of America
          </li> <li id=\\"select1--3\\" role=\\"option\\" class=\\"kindly-option\\"><i>ğŸ‡¨ğŸ‡³</i>People's Republic of China
          </li></ul></div></div>
    "
`;
